lib/circuit.js

```diff
@@ -348,6 +348,50 @@ class CircuitBreaker extends EventEmitter {
   clearCache () {
     CACHE.set(this, undefined);
   }
+
+  /**
+   * Provide a health check function to be called periodically. The function should
+   * return a Promise. If the promise is rejected, the circuit will open. This is
+   * in addition to the existing circuit behavior as defined by `options.errorThresholdPercentage`
+   * in the constructor. For example, if the health check function provided here
+   * always returns a resolved promise, the circuit can still trip and open if
+   * there are failures exceeding the configured threshold. The health check function
+   * is excecuted within the circuit breaker's execution context, so `this` within
+   * the function is the circuit breaker itself.
+   *
+   * @param func {Function} a health check function which returns a promise.
+   * @param [interval] {Number} the amount of time between calls to the health
+   * check function. Default: 5000 (5 seconds)
+   *
+   * @returns {Promise}
+   *
+   * @fires CircuitBreaker#health-check-failed
+   * @throws TypeError if `interval` is supplied but not a number
+   */
+  healthCheck (func, interval) {
+    interval = interval || 5000;
+    if (typeof func !== 'function') throw new TypeError('Health check function must be a function');
+    if (isNaN(interval)) throw new TypeError('Health check interval must be a number');
+
+    const check = _ => {
+      func.apply(this).catch(e => {
```

Comment:
related to my comment about the health-check-passed event, should we do something on non error